### PR TITLE
Fix random fails on TestPerform

### PR DESCRIPTION
**Pull request by gaiaz-iusipov**
**What this PR does / why we need it**:
Fix random fails on TestPerform caused by missing sync.Mutex

**Special notes for your reviewer**:

Also some small changes by static analysis

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -579,8 +579,8 @@ func batchPerform(infos ResourceList, fn func(*resource.Info) error, errs chan<-
 		}
 		wg.Add(1)
 		go func(i *resource.Info) {
+			defer wg.Done()
 			errs <- fn(i)
-			wg.Done()
```
Comment (willzgli) [inline pkg/kube/client.go:583]: why you make this change here ? It's seems the same effect.

Comment (gaiaz-iusipov) [reply pkg/kube/client.go:583]: This change is meaningless, but in general it's a good practice to do `wg.Done()` by `defer` so never forget to do that in any condition

Comment (gaiaz-iusipov) [reply pkg/kube/client.go]: I reverted this change

**Comment by gaiaz-iusipov** [comment]
Hello, any reviewers?

**Comment by gaiaz-iusipov** [comment]
@mattfarina can you please review this?

**Pull request closed without merging**